practice/completer/site.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/**
+ * Completer object contains completions.
+ *
+ * completions will have keys as the completion and values as weights.
+ */
+function Completer() {
+  this.completions = {};
+}
+
+/**
+ * modifyCompleter has these methods:
```

Comment:
You could probably add the specific method comments as JS docs above each method, and in this doc explain what the overall function is accomplishing.